src/main/java/com/org/jenkins/custom/jenkins/distribution/service/generators/WarGenerator.java

```diff
@@ -1,35 +1,33 @@
 package com.org.jenkins.custom.jenkins.distribution.service.generators;
 
-import com.org.jenkins.custom.jenkins.distribution.service.util.Util;
 import io.jenkins.tools.warpackager.lib.config.Config;
 import io.jenkins.tools.warpackager.lib.impl.Builder;
-
 import java.io.File;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.logging.Logger;
 
 public class WarGenerator {
 
-    private static Util util = new Util();
-
-    private static final String PACKAGER_CONFIG_YAML = "packager-config.yml";
+    private final static Logger LOGGER = Logger.getLogger(WarGenerator.class.getName());
     private static final String TEMP_PREFIX = "CDS";
 
-    public static void generateWAR(String versionName) {
+
+    public static File generateWAR(String versionName, String configuration, String artifactID) throws Exception {
+        LOGGER.info("Generating War File");
         final Config cfg;
-        Path tempDirWithPrefix = null;
-        try {
-            tempDirWithPrefix = Files.createTempDirectory(TEMP_PREFIX);
-            final File configPath = util.getFileFromResources(PACKAGER_CONFIG_YAML);
-            cfg = Config.loadConfig(configPath);
+        Path tempDirWithPrefix = Files.createTempDirectory(TEMP_PREFIX);
+            File packagerConfigFile = File.createTempFile("packager-config", ".yml");
+            byte[] buf = configuration.getBytes();
+            Files.write(packagerConfigFile.toPath(), buf);
+            cfg = Config.loadConfig(packagerConfigFile);
```

Comment:
I don't think creating a file is necessary. The Config class supports loading from an InputStream with the [load()](https://github.com/jenkinsci/custom-war-packager/blob/master/custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/config/Config.java#L58) method. The trick is to convert the string to an InputStream. This can be done with `InputStream configInputStream = new ByteArrayInputStream(configuration.getBytes());`. Maybe the encoding has to be specified in the call to `getBytes()`.

Reply:
@martinda I guess that method is private

Reply:
Ah, I missed that. You are right @sladyn98